AirCasting/SessionViews/StreamsView.swift

```diff
@@ -0,0 +1,67 @@
+// Created by Lunar on 29/06/2021.
+//
+
+import SwiftUI
+
+struct StreamsView: View {
+    
+    @Binding var selectedStream: MeasurementStreamEntity?
+    @ObservedObject var session: SessionEntity
+    var threshold: SensorThreshold
+    @EnvironmentObject private var microphoneManager: MicrophoneManager
+    let showValues: Bool
+    
+    var body: some View {
+        if session.deviceType == .MIC {
+            HStack {
+                measurementsMic
+                Spacer()
+                //This is a temporary solution for stopping mic session recording until we implement proper session edition menu
+                if microphoneManager.session?.uuid == session.uuid, microphoneManager.isRecording && (session.status == .RECORDING || session.status == .DISCONNETCED) {
+                    stopRecordingButton
+                }
+            }
+        } else {
+            ABMeasurementsView(session: session,
+                               threshold: threshold,
+                               selectedStream: _selectedStream,
+                               showValues: showValues)
+        }
+    }
+    
+    var measurementsMic: some View {
+        VStack(alignment: .leading, spacing: 5) {
+            Text("Most recent measurement:")
+            if let dbStream = session.dbStream {
+                SingleMeasurementView(stream: dbStream,
+                                      value: lastMicMeasurement(),
+                                      threshold: threshold,
+                                      selectedStream: .constant(dbStream),
+                                      showValues: showValues)
+            }
+        }
+    }
+    var stopRecordingButton: some View {
+        Button(action: {
+            try! microphoneManager.stopRecording()
+        }, label: {
+            Text("Stop recording")
+                .foregroundColor(.blue)
+        })
+    }
+    
+    func lastMicMeasurement() -> Double {
+        return session.dbStream?.latestValue ?? 0
```

Comment:
Not sure (really not sure, maybe it's ok, just pointing out) this silent unwrap fail is good here.